### PR TITLE
fix: fixed crossFade directly without play

### PR DESCRIPTION
**Pull request by luzhuang**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

**Review by johanzhu — COMMENTED** [review]
+1

**Comment by GuoLei1990** [comment]
Do a small opt with `AnimatorLayerData.ts`
```ts
switchPlayData(): void {
    const switchTemp = this.srcPlayData;
    this.srcPlayData = this.destPlayData;
    this.destPlayData = switchTemp;
  }
```

File: packages/core/src/animation/Animator.ts [hunk 1]
```diff
@@ -482,7 +482,9 @@ export class Animator extends Component {
     const { srcPlayData, destPlayData, crossFadeTransition: crossFadeTransitionInfo } = animLayerData;
     const layerAdditive = blendingMode === AnimatorLayerBlendingMode.Additive;
     const layerWeight = firstLayer ? 1.0 : weight;
-    this._checkTransition(srcPlayData, crossFadeTransitionInfo, layerIndex);
+    animLayerData.layerState !== LayerState.FixedCrossFading &&
+      this._checkTransition(srcPlayData, crossFadeTransitionInfo, layerIndex);
+
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:487]: - First of all, I want to confirm the environmental conditions that need to call `_checkTransition`, if `LayerState.CrossFading` also needs.
- Maybe we should fire the script events within range( lastFrameTime - transitionTime), and then do transition.

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**